cardano-base/src/Cardano/Base/FeatureFlags.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingStrategies #-}
+
+-- | Mini-abstraction to aid parallel development of experimental Cardano
+-- features.
+--
+-- == Usage example
+--
+-- >>> import Data.Set (Set)
+-- >>> import qualified Data.Set as Set
+-- >>> import Cardano.Base.FeatureFlags (CardanoFeatureFlag (..))
+--
+-- >>> :{
+-- logic :: Set CardanoFeatureFlag -> Int -> Int
+-- logic featureFlags
+--   | Set.member PerasFlag featureFlags = (+ 2)
+--   | otherwise = (+ 1)
+-- :}
+module Cardano.Base.FeatureFlags (
+  CardanoFeatureFlag (..),
+) where
+
+import GHC.Generics (Generic)
+import NoThunks.Class (NoThunks)
+
+-- | An experimental Cardano feature.
+data CardanoFeatureFlag
+  = -- | Feature flag for Ouroboros Peras (faster settlement).
+    PerasFlag
+  | -- | Feature flag for Ouroboros Leios (higher throughput).
+    LeiosFlag
```

Comment:
This would be my preferred order, since it is alphabetical and also reflects the importance of the feature better. But, this is very subjective, so feel free to ignore this suggestion if you do not agree
```suggestion
  = -- | Feature flag for Ouroboros Leios (higher throughput).
    LeiosFlag
  | -- | Feature flag for Ouroboros Peras (faster settlement).
    PerasFlag
```